src/ocean/time/timeout/TimeoutManager.d

```diff
@@ -228,6 +227,37 @@ abstract class TimeoutManagerBase : ITimeoutManager
 
         /***********************************************************************
 
+            Same as `unregister` but also removes expirty from list of currently
+            expired registrations. That means it can be called from `timeout`
+            callbacks, contrary to `unregister`.
+
+            Params:
+                expiry = expiry token returned by register() when registering
+                         the IExpiryRegistration instance to unregister
+
+        ***********************************************************************/
+
+        protected void drop ( ref Expiry expiry )
```

Comment:
Only thing I wonder about is how much is `unregister` useful now. Should anything call `unregister` now?

Reply:
Yes, now I remember. Feels like `unregister` should be private then, but I don't know how could we pull that off.

Reply:
I don't think so as it has to be used across ocean modules. `package(ocean)` would fix it but it is not available in D1.

Reply:
Yeah. Oh well, such is life.

Reply:
Should we make `unregister` private, in a future release? (Either as a breaking change or via a deprecation path.)

Reply:
I think this answers your last question @mihails-strasuns-sociomantic :-)

Reply:
Oh, yes, I asked the same question @gavin-norman-sociomantic , but unregister can't be private, since it needs to be accessible from other ocean internal modules - `package` would be appropriate, but it doesn't work.

Reply:
Maybe we can do some trick with `protected`, but I'd delay that for v3.x.x